src/libYARP_sig/include/yarp/sig/PointCloud.h

```diff
@@ -0,0 +1,590 @@
+                ret.addList().append((*this)(i).toBottle());
+            }
+            return ret;
+        }
+
+        /**
+         * @brief Populate the PointCloud from a yarp::os::Bottle
+         * @param[in] bt, the yarp::os::Bottle to read from. It has to be formatted in
+         * the same way it is generated by the toBottle() method.
+         * @return true for success, false otherwise
+         */
+
+        bool fromBottle(const yarp::os::Bottle& bt)
+        {
+            if (bt.isNull())
+            {
+                return false;
+            }
+
+            if (this->getPointType() != bt.get(2).asInt())
+            {
+                return false;
+            }
+
+            this->resize(bt.get(0).asInt(), bt.get(1).asInt());
+            this->header.isDense       = bt.get(3).asBool();
+            this->header.hasCustomData = bt.get(4).asBool();
+
+            if ((size_t) bt.size() != 5 + width()*height())
+            {
+                return false;
+            }
+
+            for (size_t i=0; i<this->size(); i++)
+            {
+                (*this)(i).fromBottle(bt, i+5);
+            }
+
+            return true;
+        }
+
+        virtual int getBottleTag() const override {
+            return BottleTagMap <T>();
+        }
+
+     private:
+        yarp::sig::VectorOf<T> data;
+        T nulldata;
+
+        void setPointType()
+        {
+            if (std::is_same<T, XY_DATA>::value)
+            {
+                header.pointType = PCL_POINT2D_XY;
+                return;
+            }
+
+            if (std::is_same<T, XYZ_DATA>::value)
+            {
+                header.pointType = PCL_POINT_XYZ;
+                return;
+            }
+
+            if (std::is_same<T, NORMAL_DATA>::value)
+            {
+                header.pointType = PCL_NORMAL;
+                return;
+            }
+
+            if (std::is_same<T, XYZ_RGBA_DATA>::value)
+            {
+                header.pointType = PCL_POINT_XYZ_RGBA;
+                return;
+            }
+
+            if (std::is_same<T, XYZ_I_DATA>::value)
+            {
+                header.pointType = PCL_POINT_XYZ_I;
+                return;
+            }
+
+            if (std::is_same<T, INTEREST_POINT_XYZ_DATA>::value)
+            {
+                header.pointType = PCL_INTEREST_POINT_XYZ;
+                return;
+            }
+
+            if (std::is_same<T, XYZ_NORMAL_DATA>::value)
+            {
+                header.pointType = PCL_POINT_XYZ_NORMAL;
+                return;
+            }
+
+            if (std::is_same<T, XYZ_NORMAL_RGBA_DATA>::value)
+            {
+                header.pointType = PCL_POINT_XYZ_NORMAL_RGBA;
+                return;
+            }
+
+    //        RGBA_DATA       has sense to implement them?
+    //        intensity       has sense to implement them?
+    //        VIEWPOINT_DATA  has sense to implement them?
+
+            header.pointType = 0;
+
+            return;
```

Comment:
Are those return at the end of the void methods here for some reason? I'd rather remove them, since they are useless, and they are just going to confuse the user (I think I saw another one somewhere, but it might have been this one).